File: typescript/smily.ts
/**
 * 
 * @param texts list of string [":)", ":(", ":>"]
 * @returns amount of smily face detected
 */
export function countSmilyFace(strings: string[]) : number {
    const smileyPattern = /(:[-~]?[)D]|;[-~]?[)D])/g;
    let totalCount = 0;

    strings.forEach(s => {
        const matches = s.match(smileyPattern);
        if (matches) {
            totalCount += matches.length;
        }
    });

    return totalCount;
}

File: golang/smily.go
package main

import "regexp"

func CountSmilyFace(strings []string) int {
	smileyPattern := `(:[-~]?[)D]|;[-~]?[)D])`

	re := regexp.MustCompile(smileyPattern)

	totalCount := 0

	for _, s := range strings {
		matches := re.FindAllString(s, -1)
		totalCount += len(matches)
	}

	return totalCount
}
